### ISSUE TITLE
Use podman engine config instead of exporting the variables

### ISSUE THREAD

**Issue by praveenkumar** [code-ready/crc#2926]
For podman to work with crc as of now we do `crc podman-env` which export following variables
```
export PATH="/Users/prkumar/.crc/bin/oc:$PATH"
export CONTAINER_SSHKEY="/Users/prkumar/.crc/machines/crc/id_ecdsa"
export CONTAINER_HOST="ssh://core@127.0.0.1:2222/run/user/1000/podman/podman.sock"
```

In case of podman machine it take the config from `$HOME/.config/containers/containers.conf` which have the details about connections and podman remote client read those from here without exporting `CONTAINER_SSHKEY` or `CONTAINER_HOST`

```
$ cat ~/.config/containers/containers.conf 
[containers]

[engine]
  active_service = "podman-machine-default"
  [engine.service_destinations]
    [engine.service_destinations.podman-machine-default]
      uri = "ssh://core@localhost:58456/run/user/1000/podman/podman.sock"
      identity = "/Users/prkumar/.ssh/podman-machine-default"
    [engine.service_destinations.podman-machine-default-root]
      uri = "ssh://root@localhost:58456/run/podman/podman.sock"
      identity = "/Users/prkumar/.ssh/podman-machine-default"

[machine]

[network]

[secrets]
```

**Comment by praveenkumar** [comment]
Something like this works and user can switch to different context using `podman system connection default crc` 
```
~ cat ~/.config/containers/containers.conf 
[containers]

[engine]
  active_service = "crc"
  [engine.service_destinations]
    [engine.service_destinations.crc]
      uri = "ssh://core@localhost:2222/run/user/1000/podman/podman.sock"
      identity = "/Users/prkumar/.crc/machines/crc/id_ecdsa"
    [engine.service_destinations.crc-root]
      uri = "ssh://core@localhost:2222/run/podman/podman.sock"
      identity = "/Users/prkumar/.crc/machines/crc/id_ecdsa"
    [engine.service_destinations.podman-machine-default]
      uri = "ssh://core@localhost:64940/run/user/1000/podman/podman.sock"
      identity = "/Users/prkumar/.ssh/podman-machine-default"
    [engine.service_destinations.podman-machine-default-root]
      uri = "ssh://root@localhost:64940/run/podman/podman.sock"
      identity = "/Users/prkumar/.ssh/podman-machine-default"

[machine]

[network]

[secrets]

podman system connection ls
Name                         Identity                                    URI
crc*                         /Users/prkumar/.crc/machines/crc/id_ecdsa   ssh://core@localhost:2222/run/user/1000/podman/podman.sock
crc-root                     /Users/prkumar/.crc/machines/crc/id_ecdsa   ssh://core@localhost:2222/run/podman/podman.sock
podman-machine-default       /Users/prkumar/.ssh/podman-machine-default  ssh://core@localhost:64940/run/user/1000/podman/podman.sock
podman-machine-default-root  /Users/prkumar/.ssh/podman-machine-default  ssh://root@localhost:64940/run/podman/podman.sock
```

**Comment by cfergeau** [comment]
This was addressed by 88a0aeeb5d89af1f37b4f8000a67b5c517a1a404
